app/ui-react/syndesis/public/oauth-redirect.html

```diff
@@ -0,0 +1,16 @@
+<!doctype html>
+<script>
+  let autocloseInterval;
+
+  function authCompleted() {
+    clearInterval(autocloseInterval);
+    const cred = document.cookie.split(';').filter(c => c.indexOf('cred-o2') === 0)[0];
```

Comment:
This might fail for OAuth1 (Twitter) the cookie is named `cred-o1` there. Also there could be multiple `cred-*` cookies if the user is trying multiple times to connect, picking the first one might prove to be a bit flaky.

I still can't wrap my head around the need for this, even seeing it with my own eyes that the updated cookie sent from `.../callback` is reset to the old value as soon as the popup is closed.

Reply:
Thanks for the o1/o2 thing, I wasn’t aware. There is no risk for multiple cookies because we clean ‘em all when you enter the configuration page

Reply:
And yeah, cookies and SPAs and html5 history is a dreadful combo